src/nvim/version.c

```diff
@@ -2143,24 +2146,9 @@ void list_version(void)
   MSG(longVersion);
   MSG(version_buildtype);
   list_lua_version();
+#if NVIM_INCLUDE_VERSION_CFLAGS == 1
```

Comment:
Maybe we can side-step the issue by just omitting it in Debug builds. This allows "bake-time" to see if we ever care about its absence. Then we can decide whether to remove it entirely.

```suggestion
  // Reproducible builds: omit compile info in Release builds.
#ifndef NDEBUG
```